engine/src/main/java/com/google/android/fhir/sync/upload/UploadStrategy.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.fhir.sync.upload
+
+import com.google.android.fhir.sync.upload.patch.PatchGeneratorMode
+import com.google.android.fhir.sync.upload.request.UploadRequestGeneratorMode
+import org.hl7.fhir.r4.model.codesystems.HttpVerb
+
+sealed class UploadStrategy(
```

Comment:
We do not want to allow developers to construct their own UploadStrategy. Let's ensure that by making the constructor private
```suggestion
sealed class UploadStrategy private constructor(
```

Reply:
Makes sense, the constructor for sealed class is `protected` by default.